sds/src/match_validation/config.rs

```diff
@@ -120,5 +129,28 @@ impl MatchValidationType {
 #[derive(PartialEq, Eq, Hash)]
 pub enum InternalMatchValidationType {
     Aws,
-    CustomHttp(String),
+    CustomHttp(Vec<String>),
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_http_validator_config_no_hosts() {
+        let config = HttpValidatorConfig::new("http://localhost/test", vec![]);
+        assert_eq!(config.endpoints, vec!["http://localhost/test"]);
+    }
+
+    #[test]
+    fn test_http_validator_config_with_hosts() {
+        let config = HttpValidatorConfig::new(
+            "http://localhost/$HOSTS",
+            vec!["us".to_string(), "eu".to_string()],
```

Comment:
I think we wait to raise an error is an host is specified in the endpoint but no host is available in the list and   same if there are endpoints but `$HOSTS` is missing

Reply:
done